data/transactions/logic/resources.go

```diff
@@ -223,24 +235,30 @@ func (cx *EvalContext) requireHolding(acct basics.Address, id basics.AssetIndex)
 		return nil
 	}
 	if !cx.allowsHolding(acct, id) {
-		return fmt.Errorf("unavailable Holding %s x %d would be accessible", acct, id)
+		return fmt.Errorf("unavailable Holding %d+%s would be accessible", id, acct)
 	}
 	return nil
 }
 
 func (cx *EvalContext) requireLocals(acct basics.Address, id basics.AppIndex) error {
 	if !cx.allowsLocals(acct, id) {
-		return fmt.Errorf("unavailable Local State %s x %d would be accessible", acct, id)
+		return fmt.Errorf("unavailable Local State %d+%s would be accessible", id, acct)
 	}
 	return nil
 }
 
 func (cx *EvalContext) allowsAssetTransfer(hdr *transactions.Header, tx *transactions.AssetTransferTxnFields) error {
-	err := cx.requireHolding(hdr.Sender, tx.XferAsset)
-	if err != nil {
-		return fmt.Errorf("axfer Sender: %w", err)
+	// After EnableInnerClawbackWithoutSenderHolding appears in a consensus
+	// update, we should remove it from consensus params and assume it's true in
+	// the next release. It only needs to be in there so that it gates the
+	// beahvior change in the release it first appears.
+	if !cx.Proto.EnableInnerClawbackWithoutSenderHolding || tx.AssetSender.IsZero() {
```

Comment:
Similar comment elsewhere; these 'use once and move on' params could maybe go by a special prefix or suffix..